crates/typst-library/src/foundations/str.rs

```diff
@@ -286,6 +288,28 @@ impl Str {
         Ok(c.into())
     }
 
+    /// Normalizes the string to the given Unicode Normalization Form. This is useful when
+    /// manipulating strings containing combining Unicode characters.
+    ///
+    /// ```example
+    /// #"é".normalize("NFD") \ // "e\u{0301}"
+    /// #"ſ́".normalize("NFKC") // "ś", = "\u{015b}"
```

Comment:
I'm not sure what `"ś", = "\u{015b}"` means.

Maybe this should be `typ` instead of `example` and should use an assertion instead because demonstrating the normalization visually is difficult.